### PR TITLE
:sparkles: visit object

### LINKED ISSUE #5: Feature Request: createVisitObject
First all thank you for this package, can you please add other directives implemention like 

`createVisitObject`

so that we could add directives on object location 

```js
type Query @auth {
        posts: [Post!]!
}

type Mutation @auth {
       createPost(post: CreatePostInput): Post!
}
```

### PR DESCRIPTION
**Pull request by fabienjuif**
fixes #5 

@muco-rolle is it ok for you?

I will merge and publish it in like a week if you don't respond until then.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,7 +51,10 @@
         "never"
       ],
       "import/prefer-default-export": "off",
-      "no-param-reassign": "off"
+      "no-param-reassign": "off",
```
Comment (fabienjuif) [inline package.json:54]: these are removed because we use prettier

File: src/index.js [hunk 2]
```diff
@@ -2,76 +2,72 @@
...
-        params[nextMiddleware.directiveName],
-        next,
-      )(...args)
-
-      return nextRes
+// we can modify graphql inner object with symbol making sure we don't override fields
+// right now or in futur releases
+const metaKey = Symbol('metaKey')
+
+const getResolve = ({ directiveName, params, field, middleware }) => {
```
Comment (fabienjuif) [inline src/index.js:9]: here this is a ♻️ rework  to simplify the lib

File: src/index.js [hunk 3]
```diff
@@ -2,76 +2,72 @@
...
         field,
         directiveName,
-        args: this.args,
+        params,
+        middleware,
+      })
+    }
+  }
+
+export const createVisitObject = (directiveName, middleware) =>
```
Comment (fabienjuif) [inline src/index.js:58]: the real new helper is here :)

File: package.json (hunk 1 above)
Comment (muco-rolle) [reply package.json:54]: @fabienjuif It was quite annoying to repeat a directive on every field, this is exactly what I was looking for, I spent 3 days in vain trying to achieve that. Thank you so much.

Could you please publish it.

Comment (muco-rolle) [reply package.json:54]: I was testing it, it works like a charm, and thanks the code is more readable now

Comment (fabienjuif) [reply package.json:54]: Thank you ❤️ 

I'll publish it ASAP

**Pull request merged**